pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -122,12 +122,17 @@ type PipelineResult struct {
 	// Name the given name
 	Name string `json:"name"`
 
+	// Type is the user-specified type of the result. The possible type
+	// is currently "string", "array" and "object" are alpha feature.
+	// +optional
+	Type ResultsType `json:"type,omitempty"`
```

Comment:
since `ResultsType` already has its own docstring I'm wondering if you should remove the comment about what types are supported, since if the supported types change this comment will get out of sync? If you'd like to keep it in place, I'd suggest "The possible types are 'string', 'array', and 'object', with 'string' as the default. 'array' and 'object' types are alpha features."